### PR TITLE
Add Kubermatic to pick-right-solution.md

### PR DESCRIPTION
**Pull request by chaosaffe**
Adding information for [Kubermatic](https://kubermatic.io) to Picking the Right Solution documentation

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/website/6459)

### CODE REVIEW

File: docs/setup/pick-right-solution.md [hunk 1]
```diff
@@ -59,6 +59,8 @@ a Kubernetes cluster from scratch.
 
 * [Giant Swarm](https://giantswarm.io/product/) offers managed Kubernetes clusters in their own datacenter, on-premises, or on public clouds.
 
+* [Kubermatic](https://kubermatic.io) provides managed Kubernetes clusters for various public clouds, including AWS and Digital Ocean, as well as on-premise with OpenStack integration.
+
```
Comment (steveperry-53) [inline docs/setup/pick-right-solution.md]: on-premise -> on-premises

File: docs/setup/pick-right-solution.md [hunk 2]
```diff
@@ -154,6 +158,7 @@ AppsCode.com         | Saltstack    | Debian | multi-support | [docs](https://ap
 KUBE2GO.io          |              | multi-support | multi-support | [docs](https://kube2go.io) | Commercial
 Madcore.Ai           | Jenkins DSL  | Ubuntu | flannel     | [docs](https://madcore.ai)                        | Community ([@madcore-ai](https://github.com/madcore-ai))
 Platform9        |              | multi-support | multi-support | [docs](https://platform9.com/managed-kubernetes/) | Commercial
+Kubermatic       |              | multi-support | multi-support | [docs](https://github.com/kubermatic/docs/) | Commercial
 Giant Swarm        |              | CoreOS | flannel and/or Calico | [docs](https://docs.giantswarm.io/) | Commercial
```
Comment (steveperry-53) [inline docs/setup/pick-right-solution.md]: This link returns 404.
https://github.com/kubermatic/docs/

File: docs/setup/pick-right-solution.md [hunk 3]
```diff
@@ -154,6 +158,7 @@ AppsCode.com         | Saltstack    | Debian | multi-support | [docs](https://ap
 KUBE2GO.io          |              | multi-support | multi-support | [docs](https://kube2go.io) | Commercial
 Madcore.Ai           | Jenkins DSL  | Ubuntu | flannel     | [docs](https://madcore.ai)                        | Community ([@madcore-ai](https://github.com/madcore-ai))
 Platform9        |              | multi-support | multi-support | [docs](https://platform9.com/managed-kubernetes/) | Commercial
+Kubermatic       |              | multi-support | multi-support | [docs](https://github.com/kubermatic/docs/) | Commercial
```
Comment (xiangpengzhao) [inline docs/setup/pick-right-solution.md]: Should we make all the list alphabetically ? I'm +1 on it.

Comment (chaosaffe) [reply docs/setup/pick-right-solution.md]: I think that would make the most sense. Would you like me to update the list to descending alphabetical order?

Comment (xiangpengzhao) [reply docs/setup/pick-right-solution.md]: @steveperry-53 WDYT?

Comment (steveperry-53) [reply docs/setup/pick-right-solution.md]: I don't have a strong feeling about it. But I think descending alphabetical order would be fine.

**Comment by steveperry-53** [comment]
@chaosaffe @xiangpengzhao Sorry about the delay. Are you ready for me to merge this? Or would you like to make the alphabetizing change? Thanks.

**Comment by xiangpengzhao** [comment]
@steveperry-53 I'm fine with merging it as-is if @chaosaffe has no cycles to change the order. That can be a follow-up PR. I'm not strong on changing the order either.

**Pull request merged**